ERCS/erc-7750.md

```diff
@@ -0,0 +1,103 @@
+2. **Public Accessibility**: The employment history data is publicly accessible through the `employmentHistory` mapping and can be queried using `getEmploymentHistory`.
+
+### Payment System
+1. **Salary Deposits**: Companies are required to call `depositSalary(uint contractId)` and transfer the salary amount to the smart contract's escrow. The deposited amount is tracked in an `escrowBalances` mapping.
+2. **Automated Payments**: The `releaseSalary(uint contractId)` function can be called by the contract itself (automated) or the employee token owner to release the salary according to the payment schedule. The smart contract ensures that the payment conditions (e.g., time period, contract completion) are met before transferring the funds from escrow to the employee's address.
+
+### Moderation and Dispute Resolution
+1. **Dispute Initiation**: Either party (company or employee) can initiate a dispute by calling `raiseDispute(uint contractId)`. This function triggers the assignment of a moderator to resolve the issue.
+2. **Moderator Assignment**: Moderators are assigned to each contract at the time of creation. The moderator mediates any disputes between the company and the employee, including issues such as salary payments or contract termination. Moderators can be decentralized entities or trusted third parties.
+3. **Dispute Resolution**: The assigned moderator can resolve disputes through the `resolveDispute(uint contractId, bool decisionForEmployee)` function. If the moderator rules in favor of the employee, the escrow funds are transferred to the employee token's owner. If the company's side prevails, the funds may be returned to the company.
+
+### Firing Employees
+1. **Termination Conditions**: The `terminateContract(uint contractId, string reason)` function can be called by the company if the termination conditions outlined in the contract are met. The function sets the contract status to "terminated."
+2. **Moderator's Role in Termination**: In the case of disputes regarding termination, the moderator is called upon to resolve the issue based on the contract terms and evidence provided by both parties.
+
+### Review System
+1. **Reputation Scores**: Both companies and employees can submit reviews using the `submitReview(uint contractId, uint rating, string comments)` function. These reviews are stored in a `reviews` mapping and tied to the contract ID.
+2. **Public Reviews**: Reviews can be accessed using `getReviews(uint contractId)`, which returns an array of reviews associated with a specific contract. These reviews contribute to a reputation score for both the company and employee.
```

Comment:
Generally ERCs propose a complete solidity interface, then explain the function/behaviour of its functions. While it isn't a strict requirement, I'd like to see it here.